guides/pages/how-to-add-users-to-liveblocks-comments.mdx

```diff
@@ -0,0 +1,338 @@
+---
+meta:
+  title: "How to add users to Liveblocks Comments"
+  description:
+    "Learn how to add your user’s avatars and names to Liveblocks Comments using
+    the resolver functions."
+---
+
+After following the get started guide for Comments, you’ll notice that each user
+is currently “Anonymous”, and that there’s no way to mention or tag other users.
+To enable these features, we need to tell Comments where to find your users’
+information.
+
+<Figure>
+  <Image
+    src="/assets/tutorials/comments/thread.png"
+    alt="Thread with resolved users"
+    width={1291}
+    height={892}
+  />
+</Figure>
+
+## What we’re learning
+
+In this guide we’ll be modifying
+[`createRoomContext`](/docs/api-reference/liveblocks-react#createRoomContext) in
+our config file, learning how to:
+
+- Add names and avatars to threads and comments with
+  [`resolveUsers`](/docs/api-reference/liveblocks-react#resolveUsers).
+- Create user mention suggestions in the composer using
+  [`resolveMentionSuggestions`](/docs/api-reference/liveblocks-react#resolveMentionSuggestions).
+
+## Authenticate your application
+
+The first step is to find an
+[authentication guide for your framework](/docs/rooms/authentication) and
+authenticate your app, as this is necessary for Comments.
+
+Make sure to follow the metadata step in the guide, and attach the name of your
+user, along with the URL of their avatar, as these properties will both be used
+in the [default components](/docs/products/comments#Components). Here’s an
+example using access token authentication, with an email address as a user’s ID.
+
+```ts title="Metadata in access tokens"
+const session = liveblocks.prepareSession("marc@example.com", {
+  userInfo: {
+    name: "Marc",
+    avatar: "https://example.com/marc.png",
+
+    // Your custom metadata
+    // ...
+  },
+});
+```
+
+If you’re using ID token authentication, it’ll look a little different.
+
+```ts title="Metadata in ID tokens" isCollapsed isCollapsable
+const { status, body } = await liveblocks.identifyUser(
+  {
+    userId: "marc@example.com",
+    groupIds: [], // Optional
+  },
+  {
+    userInfo: {
+      name: "Marc",
+      avatar: "https://example.com/marc.png",
+
+      // Your custom metadata
+      // ...
+    },
+  }
+);
+```
+
+Don’t forget to modify your `UserMetadata` type in `liveblocks.config.ts` to
+match the metadata format, adding type hints to your editor.
+
+```ts file="liveblocks.config.ts"
+type UserMetadata = {
+  id: string;
+
+  info: {
+    name: string;
+    avatar: string;
+
+    // Your custom metadata
+    // ...
+  };
+};
+```
+
+Any metadata you add is made saved and accessible in your Comments app with the
```

Comment:
```suggestion
Any metadata you add is saved and made accessible in your Comments app with the
```

Reply:
Also, I find this paragraph a bit confusing: users haven't learned about `resolveUsers` yet and they just saw how to declare metadata during authentication, yet here it says this data is saved and accessible via `useUser`, I think it's blurring the two concepts together.

Reply:
Agree, I'll remove the line!